Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -836,12 +851,22 @@ void getSupportedTypes(HashSet<String>& types) const final
 
 void MediaPlayerPrivateMediaSourceAVFObjC::updateDisplayLayerAndDecompressionSession()
 {
+    if (shouldEnsureVideoRenderer()) {
+        destroyDecompressionSession();
+        destroyLayer();
+        ensureVideoRenderer();
+        return;
+    }
+
     if (shouldEnsureLayer()) {
```

Comment:
I would tend to merge this into a single method shouldEnsureVideoRendererOrLayer so that we only have to deal between either decompression or the displaying mode.

Otherwise it is starting to look a bit convoluted.

But this can be done as a follow-up/fix